### ISSUE TITLE
E1121 (too-many-function-args) triggered incorrectly when metaclass __call__ accepts more arguments than class __init__

### ISSUE THREAD

**Issue by Millefeuille42** [PyCQA/pylint#11032]
### Bug description


When instantiating a class, pylint strictly checks the arguments against the class's `__init__`. However, if the class uses a metaclass that overrides `__call__` with a different signature, the metaclass's `__call__` signature should dictate the valid arguments, not `__init__`.

This results in pylint incorrectly raising E1121 on valid code.

```python
class Meta(type):
    def __call__(cls, a: int, b: int) -> "MyClass":
        return super().__call__(a)

class MyClass(metaclass=Meta):
    def __init__(self, a: int) -> None:
        self.a = a

MyClass(1, 2)
```

### Command used

```shell
pylint main.py
```

### Pylint output

```python
************* Module main                                                                                                                                                                                                                                                                                                                                                                                                                   
main2.py:12:0: E1121: Too many positional arguments for constructor call (too-many-function-args)
```

### Expected behavior

I'd expect pylint to refer to the metaclass's `__call__` instead of strictly refering to the class's `__init__`.

### Pylint version

```shell
pylint 4.0.5                                                                                                                                                                                                                                                                                                                                                                                                                                
astroid 4.0.4
Python 3.12.12 (main, Dec  9 2025, 19:05:33) [Clang 21.1.4 ]
```

### OS / Environment

macOS 26.3

**Comment by MCharming98** [comment]
I think the TypeChecker, specifically the _determine_callable function in pylint/checkers/typecheck.py, incorrectly assumes class instantiation always matches the signature of __new__ or __init__, failing to check if the class's metaclass overrides __call__ with a different signature.

Suggestion:
Modify pylint/checkers/typecheck.py: Update _determine_callable to first check if the ClassDef's metaclass overrides __call__. If a custom __call__ exists (i.e., it is not builtins.type.__call__), use its signature for argument validation.

**Comment by Millefeuille42** [comment]
Ok, thanks for the insight! I'll try to do a PR soon if that's OK?

**Comment by Millefeuille42** [comment]
Did a PR (#11055).
